ncclient/transport/ssh.py

```diff
@@ -185,8 +185,13 @@ def connect(self, host, port=830, timeout=None, unknown_host_cb=default_unknown_
             username = getpass.getuser()
 
         sock = None
-        if config.get("proxycommand"):
-            sock = paramiko.proxy.ProxyCommand(config.get("proxycommand"))
+        proxycommand = config.get("proxycommand")
+        if proxycommand:
+            if not isinstance(proxycommand, basestring):
+              proxycommand = [os.path.expanduser(elem) for elem in proxycommand]
```

Comment:
This is a paramiko internal.  I wanted to handle any iterable other than basestring in case they decide to use tuples or what.

https://github.com/paramiko/paramiko/blob/master/paramiko/config.py#L214

I'm not sure precisely how proxycommand becomes a list, but I stopped getting exceptions (re-thrown from ncclient) with this change.

I also considered asking paramiko to handle ~ internally, but then ncclient still needs to work across paramiko versions ... you can also see in that link why I removed a similar method to identityfile inside ncclient (a few lines up from here) - paramiko handles ~ internally but on a command-by-command basis, which is maddening.